src/Apps/Order/Routes/Shipping2/Components/FulfillmentDetailsForm.tsx

```diff
@@ -190,47 +193,49 @@ const FulfillmentDetailsFormLayout = (
     formikContext.submitForm()
   }
 
-  /**
-   * Effects
-   */
+  const previousFulfillmentType = usePrevious(values.fulfillmentType)
 
-  // Reset form fields when switching between ship/pickup
-  const previousValues = usePrevious(values)
+  const handleChangeFulfillmentType = async (newValue: FulfillmentType) => {
+    if (newValue === previousFulfillmentType) {
+      return
+    }
 
-  useEffect(() => {
-    const resetAttributesOnFulfillmentTypeChange = async () => {
-      if (values.fulfillmentType !== previousValues.fulfillmentType) {
-        if (values.fulfillmentType === FulfillmentType.PICKUP) {
-          await setValues({
-            ...values,
-            attributes: {
-              name: "",
-              phoneNumber: "",
-              addressLine1: "",
-              addressLine2: "",
-              city: "",
-              region: "",
-              postalCode: "",
-              country: "",
-            },
-          })
-          return
-        }
-
-        if (values.fulfillmentType === FulfillmentType.SHIP) {
-          // reset to current initialValues (based on calculation in FulfillmentDetails.tsx)
-          formikContext.resetForm()
-        }
+    shippingContext.actions.goBackToFulfillmentDetails()
+    await setTouched({})
+
+    if (newValue === FulfillmentType.PICKUP) {
+      const initialValuesForPickup: PickupValues = {
+        // ...values,
```

Comment:
Did we mean to leave those `values` out here?

Reply:
Yes - we're completely re-setting these values to empty, so there is nothing to retain here.